src/test/java/net/snowflake/client/jdbc/SnowflakeUtilTest.java

```diff
@@ -0,0 +1,143 @@
+    // when
+    SnowflakeColumnMetadata columnMetadata =
+        SnowflakeUtil.extractColumnMetadata(rootNode, false, null);
+    // then
+    assertNotNull(columnMetadata);
+    assertEquals(
+        OBJECT_MAPPER.writeValueAsString(expectedColumnMetadata),
+        OBJECT_MAPPER.writeValueAsString(columnMetadata));
+  }
+
+  private static SnowflakeColumnMetadata createExpectedMetadata(
+      JsonNode rootNode, JsonNode fieldOne, JsonNode fieldTwo) throws SnowflakeSQLLoggedException {
+    ColumnTypeInfo columnTypeInfo =
+        getSnowflakeType(rootNode.path("type").asText(), null, null, null, 0);
+    ColumnTypeInfo columnTypeInfoNodeOne =
+        getSnowflakeType(fieldOne.path("type").asText(), null, null, null, Types.BIGINT);
+    ColumnTypeInfo columnTypeInfoNodeTwo =
+        getSnowflakeType(fieldTwo.path("type").asText(), null, null, null, Types.DECIMAL);
+    SnowflakeColumnMetadata expectedColumnMetadata =
+        new SnowflakeColumnMetadata(
+            rootNode.path("name").asText(),
+            columnTypeInfo.getColumnType(),
+            rootNode.path("nullable").asBoolean(),
+            rootNode.path("length").asInt(),
+            rootNode.path("precision").asInt(),
+            rootNode.path("scale").asInt(),
+            columnTypeInfo.getExtColTypeName(),
+            false,
+            columnTypeInfo.getSnowflakeType(),
+            new FieldMetadata[] {
+              new FieldMetadata(
+                  fieldOne.path("name").asText(),
+                  columnTypeInfoNodeOne.getExtColTypeName(),
+                  columnTypeInfoNodeOne.getColumnType(),
+                  fieldOne.path("nullable").asBoolean(),
+                  fieldOne.path("length").asInt(),
+                  fieldOne.path("precision").asInt(),
+                  fieldOne.path("scale").asInt(),
+                  fieldOne.path("fixed").asBoolean(),
+                  columnTypeInfoNodeOne.getSnowflakeType(),
+                  new FieldMetadata[0]),
+              new FieldMetadata(
+                  fieldTwo.path("name").asText(),
+                  columnTypeInfoNodeTwo.getExtColTypeName(),
+                  columnTypeInfoNodeTwo.getColumnType(),
+                  fieldTwo.path("nullable").asBoolean(),
+                  fieldTwo.path("length").asInt(),
+                  fieldTwo.path("precision").asInt(),
+                  fieldTwo.path("scale").asInt(),
+                  fieldTwo.path("fixed").asBoolean(),
+                  columnTypeInfoNodeTwo.getSnowflakeType(),
+                  new FieldMetadata[0])
+            },
+            rootNode.path("database").asText(),
+            rootNode.path("schema").asText(),
+            rootNode.path("table").asText(),
+            false);
+    return expectedColumnMetadata;
+  }
+
+  private static ObjectNode createRootNode() {
+    ObjectNode rootNode = OBJECT_MAPPER.createObjectNode();
+    String name = "STRUCT";
+    rootNode.put("name", name);
+    String type = "object";
```

Comment:
do we need all those variables or we can just pass them as literals?

Reply:
ok